helper-cli/src/main/kotlin/commands/dev/RewriteTestAssetsCommand.kt

```diff
@@ -0,0 +1,204 @@
+
+import com.github.ajalt.clikt.core.CliktCommand
+import com.github.ajalt.clikt.parameters.options.convert
+import com.github.ajalt.clikt.parameters.options.option
+import com.github.ajalt.clikt.parameters.options.required
+import com.github.ajalt.clikt.parameters.types.file
+
+import java.io.File
+
+import kotlin.reflect.KClass
+
+import org.ossreviewtoolkit.model.AnalyzerResult
+import org.ossreviewtoolkit.model.AnalyzerRun
+import org.ossreviewtoolkit.model.EvaluatorRun
+import org.ossreviewtoolkit.model.FileFormat
+import org.ossreviewtoolkit.model.OrtResult
+import org.ossreviewtoolkit.model.ProjectAnalyzerResult
+import org.ossreviewtoolkit.model.ScanResult
+import org.ossreviewtoolkit.model.ScannerRun
+import org.ossreviewtoolkit.utils.common.expandTilde
+
+internal class RewriteTestAssetsCommand : CliktCommand(
+    help = "Searches all test assets directories in the given ORT sources directory for recognized serialized files " +
+            "and tries to de-serialize and serialize the file."
+) {
+    private val ortSourcesDir by option(
+        "--ort-sources-dir", "-i",
+        help = "The directory containing the source code of ORT."
+    ).convert { it.expandTilde() }
+        .file(mustExist = true, canBeFile = false, canBeDir = true, mustBeWritable = false, mustBeReadable = true)
+        .convert { it.absoluteFile.normalize() }
+        .required()
+
+    override fun run() {
+        val candidateFiles = findCandidateFiles(ortSourcesDir)
+        println("Found ${candidateFiles.size} candidate file(s).\n")
+
+        val candidateFilesByType = patchAll(candidateFiles)
+
+        candidateFilesByType.forEach { (type, files) ->
+            files.forEach { file ->
+                println("[$type] ${file.absolutePath}")
+            }
+        }
+
+        println("\nSummary\n")
+
+        candidateFilesByType.forEach { (type, files) ->
+            println("$type: ${files.size}")
+        }
+    }
+}
+
+// Fake values to replace the reaplce patterns with in order to make de-serialization work. The fake values must
+// be unique and no fake value must be a substring of another.
+private val REPLACE_PATTERN_REPLACEMENTS = listOf(
+    "<REPLACE_JAVA>" to "33445501",
+    "<REPLACE_OS>" to "33445502",
+    "\"<REPLACE_PROCESSORS>\"" to "33445503",
+    "\"<REPLACE_MAX_MEMORY>\"" to "33445504",
+    "<REPLACE_DEFINITION_FILE_PATH>" to "33445505",
+    "<REPLACE_ABSOLUTE_DEFINITION_FILE_PATH>" to "33445506",
+    "<REPLACE_URL>" to "http://replace/url/non/processed",
+    "<REPLACE_REVISION>" to "33445507",
+    "<REPLACE_PATH>" to "replace/path",
+    "<REPLACE_URL_PROCESSED>" to "http://replace/url/processed",
+)
+
+private val TARGET_CLASSES = setOf(
+    AnalyzerResult::class,
+    AnalyzerRun::class,
+    EvaluatorRun::class,
+    OrtResult::class,
+    ScannerRun::class,
+    ScanResult::class,
+    ProjectAnalyzerResult::class
+)
+
+// Paths to nodes in the tree of JsonNodes, whose subtree shall not be changed:
```

Comment:
Similar to your comment above, where you explain *why* values must be replaced (in order to make deserialization work), can you please also explain *why* these subtree must not be changed?

Reply:
I mean, I really have not clue what this (and the related code) is needed for.